plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -108,6 +110,19 @@ def associate(source_repo, dest_repo, import_conduit, config, units=None, solver
             associated_units |= tmp_associated_units
             failed_units |= tmp_failed_units
 
+    # ------ get module children of modules ------
+
+    # import pydevd_pycharm
+    # pydevd_pycharm.settrace('localhost', port=12735, stdoutToServer=True, stderrToServer=True)
+
+    modulemd_units = set(unit for unit in units if isinstance(unit, models.Modulemd))
+    modulemd_deps = get_recursive_module_deps(modulemd_units)
+    modules_to_copy = modulemd_deps | modulemd_units
+
+    # The name of this function is a lie, it copies modules too
+    associated_units |= copy_rpms(modules_to_copy, source_repo, dest_repo, import_conduit,
```

Comment:
There are really 2 ways we could approach this.  

1) Figure out what the dependencies are from the metadata and just copy them all, avoid trying to handle module dependencies with the depsolver. 

2.  Continue doing what we currently do, determining the module dependencies via the depsolver.  

I was playing around with the idea of #1 here, but we don't have to do it that way.

Reply:
One of the reasons to choose approach #2 in the past was a desire to produce dep solving results as close to dnf as possible and they use libsolv for that I think.
At the same time our use case is a bit different, we don't have info about client system and we need to copy for all the platforms.
Having said that, I think it's still fine to use approach #1 if you think that it's more manageable at the moment.